### ISSUE TITLE
Example in documentation on default iteration in incorrect (migrated from Trac #1687)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3238]
Original ticket http://projects.scipy.org/numpy/ticket/1687
Reported 2010-11-27 by atmention:gerritholl, assigned to atmention:pv.

According to

http://docs.scipy.org/doc/numpy/reference/arrays.classes.html#default-iteration

''The default behavior [of ''for val in arr''] is equivalent to:''

```
for i in arr.shape[0]:
    val = arr[i]
```

This cannot be correct, because arr.shape[0] is an int and not iterable. Correct is:

```
for i in xrange(arr.shape[0]):
    val = arr[i]
```

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2010-11-28

Fixed in commit:9273a61, thanks.

**Comment by thouis** [comment]
Comment in Trac by unknown, 2011-07-20

Milestone 1.5.2 deleted
